### DISCUSSION TITLE
Mapping between nodes and edge indices

### DISCUSSION THREAD

**Discussion by chinmay5** [pyg-team/pytorch_geometric#2588] [Q&A]
I had a small doubt and would really appreciate if someone could clarify it. In pytorch geometric, when we are loading our data, the edge index having entries such as (0,2) would comprehend it as an edge between nodes 0 and 2 in the vertex list (which we pass as data.x in our input). Is this the mapping it does? I wanted to ask this because I have nodes that are numbered but I stored them in a set. In that scenario, I should not do it since a set can reorder the entries and thereby, it would lead to wrong mappings in my input. A clarification would be really appreciated

**Comment by wsad1** [comment]
Yes that's exactly how it works. An entry like `[0,5]` in edge_index indicates that an edge exists between `x[0]` and `x[5]`. Check [this](https://pytorch-geometric.readthedocs.io/en/latest/notes/introduction.html#data-handling-of-graphs) out for a detailed example.
